tests/unit/test_asyn.py

```diff
@@ -111,6 +112,37 @@ async def process(row):
         list(mapper.iterate(timeout=4))
 
 
+@pytest.mark.parametrize("create_mapper", [AsyncMapper, OrderedMapper])
+def test_mapper_deadlock(create_mapper):
```

Comment:
will it deadlock if we don't wrap producer into a thread? were you trying to test (make sure that producer is wrapped)?